test/unit_tests/graphql/test_subscription_relations.py

```diff
@@ -0,0 +1,766 @@
+# Copyright 2022 SURF.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+from http import HTTPStatus
+
+import pytest
+
+from orchestrator.domain.base import SubscriptionModel
+from test.unit_tests.conftest import do_refresh_subscriptions_search_view
+
+
+def assert_result_ids_against_expected_ids(result_ids, expected_ids):
+    assert sorted(result_ids) == sorted([str(id) for id in expected_ids])
+
+
+def build_subscriptions_query(body: str) -> str:
+    return f"""
+query SubscriptionQuery(
+    $first: Int!,
+    $after: Int!,
+    $sortBy: [GraphqlSort!],
+    $filterBy: [GraphqlFilter!],
+    $query: String,
+    $dependsOnFilter: SubscriptionRelationFilter,
+    $dependsOnSubscriptionsFilter: [GraphqlFilter!],
+    $inUseByFilter: SubscriptionRelationFilter,
+    $inUseBySubscriptionsFilter: [GraphqlFilter!],
+) {{
+    subscriptions(first: $first, after: $after, sortBy: $sortBy, filterBy: $filterBy, query: $query)
+    {body}
+}}
+"""
+
+
+def get_subscriptions_query_with_relations(
+    first: int = 10,
+    after: int = 0,
+    filter_by: list[dict[str, str]] | None = None,
+    sort_by: list[dict[str, str]] | None = None,
+    query_string: str | None = None,
+    depends_on_filter: dict[str, str] | None = None,
+    depends_on_subscription_filter: dict[str, str] | None = None,
+    in_use_by_filter: dict[str, str] | None = None,
+    in_use_by_subscription_filter: dict[str, str] | None = None,
+) -> bytes:
+    query = build_subscriptions_query(
+        """{
+    page {
+      description
+      subscriptionId
+      status
+      insync
+      note
+      startDate
+      endDate
+      productBlockInstances {
+        ownerSubscriptionId
+        inUseByRelations
+      }
+      processes(sortBy: [{field: "startedAt", order: ASC}]) {
+        page {
+          processId
+          isTask
+          lastStep
+          lastStatus
+          assignee
+          failedReason
+          traceback
+          workflowName
+          createdBy
+          startedAt
+          lastModifiedAt
+          product {
+            productId
+            name
+            description
+            productType
+            status
+            tag
+            createdAt
+            endDate
+          }
+        }
```

Comment:
Are these 2 blocks relevant for the test? I see that you're inspecting productBlockInstances in one testcase as an extra check, but I think just verifying the end result (as in, which related subscriptions are returned) is sufficient here. So maybe remove productBlocklnstances/processes (the file is long enough already :) )

Reply:
there is a test for `test_single_subscription_with_processes` which uses the same query function, but will split it up so only the processes test fetches the processes.